src/erl_optics_foil_server.erl

```diff
@@ -0,0 +1,114 @@
+-module(erl_optics_foil_server).
+-include("erl_optics.hrl").
+
+-behaviour(gen_server).
+
+-export([
+    init/1,
+    handle_cast/2,
+    handle_call/3,
+    handle_info/2,
+    terminate/2,
+    code_change/3
+]).
+
+-export([
+    start_link/0,
+    add_lens/2,
+    stop/0
+]).
+
+-define(SERVER, ?MODULE).
+
+-record(state, {
+    foil_namespace :: ?NS,
+    new_lenses  :: map(),
+    timer_ref :: timer:tref() | undefined
+}).
+
+%%%=========
+%%% API
+%%%=========
+
+%Modes: carbon | blank
+
+start_link() ->
+    gen_server:start_link({local, ?SERVER}, ?MODULE, [], []).
+
+stop() ->
+    gen_server:cast(?SERVER, stop).
+
+%%%==========
+%%% Callbacks
+%%%==========
+
+%% TODO: Change the timer to start after a new lens has been
+%%       declared and stop after a while to avoid a permanent
+%%       server doing useless things.
+
+init(_) ->
+    Interval = ?ENV(?ENV_FOIL_RELOAD_INTERVAL, ?DEFAULT_FOIL_RELOAD_INTERVAL),
+    {ok, Tref} = timer:send_interval(Interval, reload_foil),
```

Comment:
I've thought about this some more, and I'm thinking that it might be better to have this gen_server store and compare timestamps. This is because I think that comparing timestamps will lead to better behavior if/when this server is being spammed with messages. With `timer`, all messages before the message from `timer` will have to be handled, but that's not the case if the gen_server stores and compares timestamps.

Do you think this server will become overloaded when rtb_gateway starts?

Reply:
With the ets check that has been added, the server should not be spammed. It will only receive one message per new lens.